### PR TITLE
Some tests for versions_check_set.sh

### PR DESCRIPTION
**Pull request by danpoltawski**
Not exhasitive! There are a lot of things done by this script, more
tests can be added.

#114 Adds a setup test for GNU style date which is needed for this script to run properly.

### CODE REVIEW

File: tests/99-versions_check_set.bats [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    assert_output --partial "ERROR: No correct version (10 digits + opt 2 more) found"
+}
+
+@test "versions_check_set/versions_check_set.sh: version too far in the future" {
+    git_apply_fixture versions_check_set/version_too_far_in_future.patch
+
+    ci_run versions_check_set/versions_check_set.sh
+    assert_failure
+    run grep ERROR $WORKSPACE/versions_check_set.txt
+    assert_output --partial "ERROR: No correct actual (<+7d) date found (2716-05-23)"
```
Comment (stronk7) [inline tests/99-versions_check_set.bats]: Pity this will run only till 2716!

**Comment by stronk7** [comment]
Using my admin superpowers, because the branch was not rebased on top of current master. Maybe superpowers should not required for that?

**Comment by danpoltawski** [comment]
> Maybe superpowers should not required for that?

Isn't there a button to rebase the branch which doens't need you to use that?

**Comment by stronk7** [comment]
Will (try to remember to) look next time, have not noticed it, thanks for the tip!

**Pull request merged**